plugin/storage/kafka/writer.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"github.com/Shopify/sarama"
+	"github.com/uber/jaeger-lib/metrics"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+type spanWriterMetrics struct {
+	SpansWrittenSuccess metrics.Counter
+	SpansWrittenFailure metrics.Counter
+}
+
+// SpanWriter writes spans to kafka. Implements spanstore.Writer
+type SpanWriter struct {
+	metrics    spanWriterMetrics
+	producer   sarama.AsyncProducer
```

Comment:
is this an interface or struct?  we should depend on interface, otherwise you cannot test this file

Reply:
It's an interface. https://github.com/Shopify/sarama/blob/35324cf48e33d8260e1c7c18854465a904ade249/async_producer.go#L19